src/System.Buffers/src/System/ByteSpan.cs

```diff
@@ -74,7 +75,8 @@ public int Length
 
         public ByteSpan Slice(int index)
         {
-            Precondition.Require(index < Length);
+            if (index >= Length)
+                return new ByteSpan(_data, 0);
```

Comment:
It's clearly an error then. We should either throw an exception or use Try pattern, if perf is an issue (I would just throw for now).